BraintreeUIKit/Models/BTUIKCardType.m

```diff
@@ -84,17 +84,9 @@ + (NSArray *)possibleCardTypesForNumber:(NSString *)number {
         return [[self class] allCards];
     }
     NSMutableSet *possibleCardTypes = [NSMutableSet set];
-    for (BTUIKCardType *cardType in [[self class] allCards]) {
-        for (NSString *prefix in cardType.validNumberPrefixes) {
-            NSUInteger compareLength = MIN(prefix.length, number.length);
-
-            NSString *sizedPrefix = [prefix substringToIndex:compareLength];
-            NSString *sizedNumber = [number substringToIndex:compareLength];
-            if ([sizedNumber isEqualToString:sizedPrefix]) {
-                [possibleCardTypes addObject:cardType];
-                break;
-            }
-        }
+    BTUIKCardType *card = [self cardTypeForNumber: number];
+    if (card) {
+        [possibleCardTypes addObject:card];
```

Comment:
I think some of the card bins could overlap each other. However, if the method isn't currently really used, I think the change you suggested could be done and if we need to go back and add more valid cards, we could bring it back if it makes maintenance easier.